src/test/java/it/pagopa/transactions/commands/handlers/TransactionUserCancelHandlerTest.java

```diff
@@ -0,0 +1,166 @@
+public class TransactionUserCancelHandlerTest {
+
+    private TransactionUserCancelHandler transactionUserCancelHandler;
+    @Mock
+    private TransactionsEventStoreRepository<Void> transactionEventUserCancelStoreRepository;
+
+    @Mock
+    private QueueAsyncClient transactionUserCancelQueueClient;
+
+    private TransactionsEventStoreRepository<Object> eventStoreRepository = Mockito
+            .mock(TransactionsEventStoreRepository.class);
+    private final TransactionsUtils transactionsUtils = new TransactionsUtils(eventStoreRepository);
+
+    @BeforeEach
+    private void init() {
+        transactionUserCancelHandler = new TransactionUserCancelHandler(
+                transactionEventUserCancelStoreRepository,
+                transactionUserCancelQueueClient,
+                transactionsUtils
+        );
+    }
+
+    @Test
+    void shouldSaveAuthorizationEvent() {
+        String transactionId = TransactionTestUtils.TRANSACTION_ID;
+        TransactionUserCancelCommand transactionUserCancelCommand = new TransactionUserCancelCommand(
+                null,
+                new TransactionId(UUID.fromString(transactionId))
+        );
+
+        /* PRECONDITION */
+        Mockito.when(eventStoreRepository.findByTransactionId(transactionId))
+                .thenReturn((Flux) Flux.just(TransactionTestUtils.transactionActivateEvent()));
+
+        Mockito.when(transactionEventUserCancelStoreRepository.save(any()))
+                .thenAnswer(a -> Mono.just(a.getArgument(0)));
+
+        Mockito.when(transactionUserCancelQueueClient.sendMessageWithResponse(any(BinaryData.class), any(), any()))
+                .thenReturn(queueSuccessfulResponse());
+        /* EXECUTION TEST */
+        StepVerifier.create(transactionUserCancelHandler.handle(transactionUserCancelCommand))
+                .consumeNextWith(
+                        next -> {
+                            assertEquals(TransactionEventCode.TRANSACTION_USER_CANCELED_EVENT, next.getEventCode());
+                        }
+                )
+                .verifyComplete();
+    }
+
+    @Test
+    void shouldSaveAuthorizationEventWithError() {
+        String transactionId = TransactionTestUtils.TRANSACTION_ID;
+        TransactionUserCancelCommand transactionUserCancelCommand = new TransactionUserCancelCommand(
+                null,
+                new TransactionId(UUID.fromString(transactionId))
+        );
+
+        /* PRECONDITION */
+        Mockito.when(eventStoreRepository.findByTransactionId(transactionId))
+                .thenReturn((Flux) Flux.just(TransactionTestUtils.transactionActivateEvent()));
+
+        Mockito.when(transactionEventUserCancelStoreRepository.save(any()))
+                .thenReturn(Mono.error(new RuntimeException()));
+
+        /* EXECUTION TEST */
+        StepVerifier.create(transactionUserCancelHandler.handle(transactionUserCancelCommand))
+                .expectError(RuntimeException.class)
+                .verify();
```

Comment:
Same here, here maybe the verify assertion can be that queue client is called 0 times